esp/smc/SMCLib/TpCommon.cpp

```diff
@@ -170,6 +170,16 @@ extern TPWRAPPER_API bool validateDropZoneHostAndPath(const char* dropZoneName,
     Owned<IPropertyTree> plane = getDropZonePlane(dropZoneName);
     if (nullptr == plane)
         return false;
+    if (isContainerized() && streq("localhost", hostToCheck))
+    {
+        //For historical reasons, ESP service returns "localhost" for planes without hosts
+        //in k8s mounted dropzones. When the "localhost" is received for a plane, avoid 
+        //treating this "localhost" usage as invalid.
+        StringBuffer host;
+        if (!getPlaneHost(host, plane, 0))
```

Comment:
not sure why coded like this.
Isn't the new code saying:

If = "localhost" then get 1st host of plane (so not a mounted plane) - and since this is containerized, "localhost" is not a legal possibility.
Or, if can't get host (because no hosts, because it's a mounted plane), then use null, i.e. don't check host.

Why not just:
```
if (isContainerized() && streq("localhost", hostToCheck))
    hostToCheck = nullptr;
```